docs/commerce/checkout/customize_checkout.md

```diff
@@ -0,0 +1,202 @@
+
+- it is based on Symfony workflow
+- exposes a variety of APIs
+- exposes Twig functions that help you render the steps
+
+The most important contract exposed by the package is the `CheckoutServiceInterface` interface. 
+It exposes a number of methods that you can call, for example, to load checkouts based 
+on checkout identifier or for a specific cart. 
+Other methods help you create, update, or delete checkouts. 
+
+For more information, see [Checkout API](checkout_api.md).
+
+## Add checkout stage
+
+By default [[= product_name =]] comes with a multi-step checkout process, which you can expand by adding steps.
+For example, if you were creating a project for selling theater tickets, you could add a step 
+that allows users to select their seats.
+
+### Define workflow
+
+You can create workflow definitions in the `config/packages/ibexa.yaml` file. 
+Each workflow definition consists of a series of stages as well as a series of transitions between the stages. 
+
+To create a new workflow, for example, `seat_selection_checkout`, modify the default workflow that comes with the storefront module, by adding a  `select_seat` stage.
+
+``` yaml hl_lines="3 15"
+[[= include_file('code_samples/front/shop/checkout/config/packages/ibexa.yaml', 25, 27) =]] [[= include_file('code_samples/front/shop/checkout/config/packages/ibexa.yaml', 103, 120) =]] [[= include_file('code_samples/front/shop/checkout/config/packages/ibexa.yaml', 23, 24) =]]
+```
+
+Then, modify a list of transitions. 
+When defining a new transition, within its metadata, map the transition to its controller, and set other necessary details, such as the next step and label.
+
+``` yaml hl_lines="2 12"
+[[= include_file('code_samples/front/shop/checkout/config/packages/ibexa.yaml', 120, 133) =]] [[= include_file('code_samples/front/shop/checkout/config/packages/ibexa.yaml', 23, 24) =]]
+```
+
+### Create controller
+
+At this point you must add a controller that supports the newly added step.
+In this case, you want users to select seats in the audience.
+
+In the `src/Controller/Checkout/Step` folder, create a file that resembles the following example.
+
+The controller contains a Symfony form that collects user selections. 
+It can reuse fields and functions that come from the checkout component, for example, 
+after you check whether the form is valid, use the `advance` method to go to the next stage 
+of the process.
+
+``` php hl_lines="23 24"
+[[= include_file('code_samples/front/shop/checkout/src/Controller/SelectSeatStepController.php') =]]
+```
+
+### Create Twig template
+
+You also need a Twig template to render the Symfony form.
+In `templates/themes/custom/storefront/checkout/step`, create a layout that uses JavaScript to translate clicking into a grid to a change in value:
+
+```html+twig
+[[= include_file('code_samples/front/shop/checkout/templates/themes/storefront/checkout/step/select_seat.html.twig') =]]
+```
+
+### Configure repository
```

Comment:
I'd change the heading, you don't really configure the repository here, but rather select which workflow is used by the repository